data_hub_api/docmaps/v2/codecs/elife_manuscript.py

```diff
@@ -67,11 +67,12 @@ def get_docmap_elife_manuscript_output(
 
 
 def get_elife_manuscript_volume(
-    manuscript_version: ApiManuscriptVersionInput
+    first_manuscript_version: ApiManuscriptVersionInput
 ) -> Optional[str]:
-    rp_publication_year = manuscript_version['rp_publication_year']
-    if rp_publication_year and rp_publication_year >= ELIFE_FIRST_PUBLICATION_YEAR:
-        return str(rp_publication_year - ELIFE_FIRST_PUBLICATION_YEAR)
+    if first_manuscript_version['rp_publication_timestamp']:
+        first_rp_publication_year = first_manuscript_version['rp_publication_timestamp'].year
+        if first_rp_publication_year >= ELIFE_FIRST_PUBLICATION_YEAR:
```

Comment:
Do we need to have the check? Or can we make it an assert instead?

Reply:
> Do we need to have the check? Or can we make it an assert instead?

It think we need to check as there can be a typo as it will be filled manually. And how to check depends on behaviour we want to achieve. If we want the API not to respond when the year is earlier than 2011, we can implement an assert. Alternatively, if we're okay with having a null value in the volume, we can leave it as is.

Reply:
Okay, could point. Maybe could check with the production team. If the API is failing it makes it obvious that something is wrong. Whereas a `null` value could potentially silently be ignore. Although I guess the production team requires it and it shouldn't be ignored.